[#2] User Entity 설계

Jasonarea opened the pull request:
1. User 엔티티 생성 (id, email, password, nickname, createdAt 등)
2. PostgreSQL 연결
3. DB 헬스 체크 컨트롤러 추가
4. Security Config에 permitAll 설정  -> 추후 JWT 인증 도입 예정

* 실행가이드
docker run -d --name vibeup-db -e POSTGRES_DB=vibeup -e POSTGRES_USER=vibeup -e POSTGRES_PASSWORD=vibeupjason123 -p 5432:5432 postgres:15

* 헬스체크 완료
curl -i http://localhost:8080/health/db
<img width="723" height="216" alt="image" src="https://github.com/user-attachments/assets/c3450863-7d4c-4e7c-bd26-972a6b5c3681" />

Jasonarea committed:
feat: Add User entity and repository (#2)

Jasonarea committed:
chore: configure PostgreSQL and add DB health check (#2)

Jasonarea committed:
feat: build.gradle flyway-postgresql 추가 (#2)

Jasonarea committed:
Spring Security Config 추가 (permit All 설정)(#2)

f-lab-jason commented on src/main/java/com/flab/vibeup/health/DbHealthController.java:
```diff
@@ -0,0 +1,23 @@
+
+@RestController
+@RequestMapping("/health")
+public class DbHealthController {
```
nit: DB health checker 같은 경우는 spring boot에 build-in으로 설정되어있는것도 있구요 (`/actuator/health`로 호출할 수 있습니다)
클래스로 만드실거면 보통 JPA를 쓰는경우 ds를 직접 호출하는건 선호되진 않습니다. (connection 관리 같은 것 때문에요)
아래와 같이 더 간단한 방법이 좋을 것 같아요 🙂 
```
@RestController
public class DbHealthController {

    @PersistenceContext
    private EntityManager em;

    @GetMapping("/health/db")
    public String ping() {
        Integer result = (Integer) em.createNativeQuery("select 1").getSingleResult();
        return "DB OK: " + result;
    }
}
```

Jasonarea replied:
피드백 감사합니다.

f-lab-jason reviewed (approved):
전체적으로 잘 세팅하셨네요! 🙌

Jasonarea committed:
feat: DB HealthCheck 개선 및 dev 환경 분리 설정 추가(#2)
- 기존 DataSource 직접 호출 방식 대신 EntityManager 기반 health check로 변경
  → JPA 환경에 더 적합하고 connection 관리 측면에서 안정적임
- Spring Boot Actuator health (/actuator/health) 활용 가능하도록 구성
- application.yml을 application-dev.yml로 분리하여 개발 환경 전용 설정 관리
